rcl_lifecycle/test/test_rcl_lifecycle.cpp

```diff
@@ -133,14 +133,19 @@ TEST(TestRclLifecycle, lifecycle_transition) {
 
   ret = rcl_lifecycle_transition_init(
     &transition, expected_id, &expected_label[0], nullptr, nullptr, &allocator);
-  EXPECT_EQ(ret, RCL_RET_OK);
-  rcutils_reset_error();
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  ret = rcl_lifecycle_transition_fini(&transition, &allocator);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
 
   ret = rcl_lifecycle_transition_init(
     &transition, expected_id, &expected_label[0], start, nullptr, &allocator);
-  EXPECT_EQ(ret, RCL_RET_OK);
-  rcutils_reset_error();
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  ret = rcl_lifecycle_transition_fini(&transition, &allocator);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
 
+  start = reinterpret_cast<rcl_lifecycle_state_t *>(
+    allocator.allocate(sizeof(rcl_lifecycle_state_t), allocator.state));
+  *start = rcl_lifecycle_get_zero_initialized_state();
```

Comment:
This is changing the semantics of the test though. We expect the `start` state to be `fini`d as well in the call to `rcl_lifecycle_transition_fini`.

Reply:
Below is original codes

https://github.com/ros2/rcl/blob/944068f3d99d6a4b1130fd62e38c69b96f0e759e/rcl_lifecycle/test/test_rcl_lifecycle.cpp#L139-L148

As you know, I add rcl_lifecycle_state_machine_fini() at line 143, which lead that `start` is freed at rcl_lifecycle_state_machine_fini(). In order to keep original logic, I have to reallocate `start` before next test.

Reply:
you're absolutely right! Thanks for the clarification.